src/main/java/com/example/bookstoreapp/exception/BookApiException.java

```diff
@@ -0,0 +1,11 @@
+package com.example.bookstoreapp.exception;
+
+import java.time.LocalDateTime;
+import org.springframework.http.HttpStatus;
+
+public record BookApiException(
```

Comment:
this is misleading

it seems that this class claims to be an exception, although it is not

it is rather `ErrorResponseWrapper`